### DISCUSSION TITLE
Missing previews for a few system apps

### DISCUSSION THREAD

**Discussion by neilswann80** [eduardofilo/RG350_adam_image#200] [SimpleMenu topics]
@eduardofilo I spotted the preview images are missing for `Terminal`, `System Info` and `Passwd`. Had a little poke around and noticed Simplemenu isn't playing nicely with `.fgl' files.

It seems `.png` images in the `.previews` media folder aren't corresponding (as they should) with the `.fgl` files, instead they appear linked to the `exec` command in the `.fgl` itself... weird!

Gets really weird when there is a `params` line also.

For example for `passwd.fgl`:
exec=/usr/bin/sudo
params= -S od-passwd-config od

If you rename the image file in previews: 
"sudo -S od-passwd-config od.png"

it will actually show the preview!  :/

I've raised this weird behavior as an issue with @FGL82 at his Simplemenu repository.
Not really an urgent issue, but I'm really curious as to what's causing it.

EDIT:  I've tried using an .fbl to link to an .fgl... even a .fgl to another .fgl, or a .fbl direct to the exec... none worked.  :(

**Comment by eduardofilo** [comment]
Those three launchers are new in 1.4 since SimpleMenu 10.3 supports launching applications with complex callbacks like the one you mention. I was testing and couldn't get the previews to show up. I didn't get as far as you did. I'll take a note for a future version.

While we're on the subject, I've been meaning to ask you how you figured out the mechanism for the fbl's that made it possible to improve Quake's listings.

**Reply by neilswann80** [reply]
In all honesty it was just a lot of trial and error experimentation.  I think the file extension could be pretty much anything within reason, but I just picked something that didn't seem to already have a purpose and was similar to `fgl` to make it seem logical to everyone.  I knew the `fgl` files themselves were a special category, but noticed the `svm` files for SCUMMVM could be used for friendly naming and just played around a bit with the concept.

Definitely a happy accident over any genius on my part!
